### ISSUE TITLE
Disabled LDAP users can still use kubectl

### ISSUE THREAD

**Issue by Typositoire** [rancher/rancher#30862]
**What kind of request is this (question/bug/enhancement/feature request):** Question / Potential Bug

**Steps to reproduce (least amount of steps as possible):** 

- Setup LDAP authentication
- Disable user in LDAP
- run kubectl get pods

**Result:**
I'm able to list pods but not login to the Rancher UI

UI shows my user with an error: `Unable to fetch user info`
In the logs we see: `[rancher-adgear-6cff8d5d85-cbgqc] 2021/01/15 13:59:16 [ERROR] Unknown error: Permission denied`

**Other details that may be helpful:**
If I disable the user IN rancher it works as expected, I cannot use kubectl nor the Rancher UI. I would expect disabled users to be sync'd in Rancher.

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): v2.5.1
- Installation option (single install/HA): HA

**Comment by vincent99** [comment]
TL;DR wait longer, make another request a few seconds later, or additionally directly disable the Rancher user if you are worried about what that user might do in a limited time window.

Kubectl access is controlled via API key.  Keys use a cache of the auth info for the underlying user because it would be unbearably slow to ask LDAP "hey is Typositoire still a user? What groups are they in?" on _every_ API call, multiple times a second.

The default cache interval is an hour (`auth-user-info-max-age-seconds` setting).  If the info for your user is older than that,  an update is triggered in the background. Your request uses the existing (now stale) info, so that it is not blocked waiting for LDAP to respond (or broken if LDAP is unreachable).

The non-blocking part of that leaves a small window for a request or two from a deleted user to get through, days after they're deleted it someone is clever.  So there's a separate setting to periodically synchronize _everyone_ on a schedule, even if there have been no API calls.  That defaults to daily (`auth-user-info-resync-cron`) at midnight UTC.

When you try to do an interactive login, that is different, we don't know your password so it is checked against the auth provider immediately (and blocking).

**Comment by Typositoire** [comment]
Oh cool, we did try and wait 15 minutes but I guess an hour is good. Sorry for that then :/ It also makes sense. So just to be sure, if I disable a user in LDAP, within an hour the user should show up disabled in Rancher because it's stale ? If that is correct, I'm happy.

**Comment by Typositoire** [comment]
Just saw your edit. Thanks! I'll close this issue and sorry again for the noise.

**Comment by Typositoire** [comment]
So we tried disabling a user and even after a whole weekend kubectl still works and user didn't sync up disabled status in rancher. Any idea what could be happening ?

**Comment by Martin-Weiss** [comment]
Might this issue be a security issue? Does this still exist in Rancher 2.5.8?

**Comment by dsim62** [comment]
Hi,

Yes we are still facing this issue on Rancher 2.5.8, on one of our customers, we disabled or deleted users in active directory but there are still active in rancher after days

**Comment by dpiraud** [comment]
> Hi,
> 
> Yes we are still facing this issue on Rancher 2.5.8, on one of our customers, we disabled or deleted users in active directory but there are still active in rancher after days

Up for this comment, facing the same issue

**Comment by Martin-Weiss** [comment]
unstale

**Comment by icanhazbeer** [comment]
Hi,
Was this ever figured out?
